File: mvp-java/domain/src/main/java/com/wasisto/githubuserfinder/domain/usecases/GetUserUseCase.java
package com.wasisto.githubuserfinder.domain.usecases;

import com.wasisto.githubuserfinder.domain.interfaces.GithubDataSource;
import com.wasisto.githubuserfinder.domain.models.User;

public class GetUserUseCase {

    private GithubDataSource githubDataSource;

    public GetUserUseCase(GithubDataSource githubDataSource) {
        this.githubDataSource = githubDataSource;
    }

    public User execute(String username) throws Exception {
        return githubDataSource.getUser(username);
    }
}

File: mvvm-kotlin/domain/src/main/java/com/wasisto/githubuserfinder/domain/usecases/GetUserUseCase.kt
package com.wasisto.githubuserfinder.domain.usecases

import com.wasisto.githubuserfinder.domain.interfaces.GithubDataSource

class GetUserUseCase(private val githubDataSource: GithubDataSource) {
    operator fun invoke(username: String) = githubDataSource.getUser(username)
}